function-service-impl/src/main/java/com/future/function/service/impl/feature/scoring/QuizServiceImpl.java

```diff
@@ -80,20 +81,24 @@ public Quiz copyQuizWithTargetBatchCode(String targetBatchCode, Quiz quiz) {
   public Quiz createQuiz(Quiz request) {
 
     return Optional.ofNullable(request)
-      .map(this::setBatchAndQuestionBank)
+      .map(this::setBatch)
+      .map(this::setQuestionBank)
       .map(quizRepository::save)
       .orElseThrow(() -> new UnsupportedOperationException(
         "Failed on #createQuiz #QuizService"));
   }
 
-  private Quiz setBatchAndQuestionBank(Quiz quiz) {
+  private Quiz setQuestionBank(Quiz quiz) {
 
-    quiz.setBatch(batchService.getBatchByCode(quiz.getBatch()
-                                                .getCode()));
     quiz.setQuestionBanks(getQuestionBanksFromService(quiz.getQuestionBanks()));
     return quiz;
   }
 
+  private Quiz setBatch(Quiz quiz) {
```

Comment:
Make this reusable, send the batch and quiz as parameter, then reuse at line 71

Reply:
already reusable to 2 method: create and update quiz, if i want to update it to receive 2 parameters, it will have to change those 2 methods and make the chaining ugly

Reply:
Ok noprob